### PR TITLE
Pipeline Improvements

### PR DESCRIPTION
**Pull request by riedgar-ms**
Makes a number of improvements to the pipeline YAMLs:
* Common template for installing Fairlearn
* Common test template for unit tests and notebooks

The first of these enables the previously separate test templates for 'regular' builds and the release pipeline to be consolidated. The second enables the templates for unit and notebook tests to be combined. As a result, several YAML files can be deleted.

This is part of #371 but further changes are required in order to close that issue.

### CODE REVIEW

File: devops/templates/all-tests-job-template.yml [hunk 1]
```diff
@@ -42,12 +75,27 @@ jobs:
...
     displayName: "Run flake8"
 
-  - script: python -m pytest test/ --ignore=test/install -m "not notebooks" --junitxml=./TEST--TEST.xml -o junit_suite_name="$(Agent.JobName)"
-    displayName: 'Run tests'
+  # =================================================
+  - ${{ if eq(parameters.testRunType, 'Unit')}}:
+    - script: python -m pytest test/ --ignore=test/install -m "not notebooks" --junitxml=./TEST--TEST.xml -o junit_suite_name="$(Agent.JobName)"
+      displayName: 'Run unit tests'
+
+  - ${{ if eq(parameters.testRunType, 'Notebooks')}}:
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply devops/templates/all-tests-job-template.yml:95]: No, they're selected by `testRunType`

File: devops/templates/fairlearn-installation-step-template.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+- ${{ if notIn(parameters.installationType, 'None', 'Editable', 'PyPI') }}:
+  - script: exit 1
+    displayName: "Bad installationType: ${{parameters.installationType}}"
+    condition: notIn('${{parameters.installationType}}', 'None', 'Editable', 'PyPI')
+
+# ==============================================================
+# Install editable
+- ${{ if eq(parameters.installationType, 'Editable') }}:
+  - script: pip install -e .
+    displayName: "Install fairlearn in editable mode"
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply devops/templates/fairlearn-installation-step-template.yml:30]: Would you prefer 'PipLocal' and drop the `-e`?

[the remark this answers is not in the document]
Comment (romanlutz) [reply devops/templates/fairlearn-installation-step-template.yml:30]: Sure. Editable mode sort of indicates that were changing things, which to my knowledge we aren't. Also, the dashboard doesn't get installed properly if you do -e

[the remark this answers is not in the document]
Comment (riedgar-ms) [reply devops/templates/fairlearn-installation-step-template.yml]: Changed

**Pull request merged**